File: app/src/main/java/droiddevelopers254/droidconke/adapters/SpeakersAdapter.java
package droiddevelopers254.droidconke.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
import com.bumptech.glide.request.RequestOptions;

import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;
import droiddevelopers254.droidconke.R;
import droiddevelopers254.droidconke.models.SpeakersModel;

public class SpeakersAdapter extends RecyclerView.Adapter<SpeakersAdapter.MyViewHolder> {
    private List<SpeakersModel> speakersList;
    private Context context;

    public SpeakersAdapter(List<SpeakersModel> speakersList,Context context){
        this.speakersList=speakersList;
        this.context=context;
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        TextView speakerNameText,speakerCompanyText;
        CircleImageView speakerImg;
        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            speakerNameText=itemView.findViewById(R.id.speakerNameText);
            speakerCompanyText=itemView.findViewById(R.id.speakerCompanyText);
            speakerImg=itemView.findViewById(R.id.speakerImg);
        }
    }
    @NonNull
    @Override
    public SpeakersAdapter.MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.speaker_details, parent, false);
        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull SpeakersAdapter.MyViewHolder holder, int position) {
        SpeakersModel speakersModel=speakersList.get(position);
        holder.speakerNameText.setText(speakersModel.getName());
        holder.speakerCompanyText.setText(speakersModel.getCompany());

        //load speaker profile image
        Glide.with(context).load(speakersModel.getPhotoUrl())
                .thumbnail(Glide.with(context).load(speakersModel.getPhotoUrl()))
                .transition(new DrawableTransitionOptions()
                        .crossFade())
                .apply(new RequestOptions()
                        .centerCrop()
                        .diskCacheStrategy(DiskCacheStrategy.ALL))
                .into(holder.speakerImg);

    }

    @Override
    public int getItemCount() {
        return speakersList.size();
    }

}

File: app/src/main/java/droiddevelopers254/droidconke/adapters/SpeakersAdapter.kt
package droiddevelopers254.droidconke.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.bumptech.glide.request.RequestOptions
import droiddevelopers254.droidconke.R
import droiddevelopers254.droidconke.models.SpeakersModel
import kotlinx.android.synthetic.main.speaker_details.view.*

class SpeakersAdapter(private val speakersList: List<SpeakersModel>, private val context: Context) : RecyclerView.Adapter<SpeakersAdapter.MyViewHolder>() {

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var speakerNameText = itemView.speakerNameText
        internal var speakerCompanyText = itemView.speakerCompanyText
        internal var speakerImg = itemView.speakerImg

       fun bindSpeakerDetails(speakersModel: SpeakersModel){
           with(speakersModel){
               Glide.with(itemView.context).load(photoUrl)
                       .thumbnail(Glide.with(itemView.context).load(photoUrl))
                       .apply(RequestOptions()
                               .centerCrop()
                               .diskCacheStrategy(DiskCacheStrategy.ALL))
                       .transition(DrawableTransitionOptions()
                               .crossFade())
                       .into(speakerImg)

               speakerNameText.text = name
               speakerCompanyText.text = company

           }
       }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SpeakersAdapter.MyViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.speaker_details, parent, false)
        return MyViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: SpeakersAdapter.MyViewHolder, position: Int) {
     holder.bindSpeakerDetails(speakersList[position])
    }

    override fun getItemCount(): Int {
        return speakersList.size
    }

}
